pkg/authz/capabilities.go

```diff
@@ -0,0 +1,139 @@
+package authz
+
+import (
+	"net/http"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+)
+
+// CapabilityChecker is responsible for checking if a user has the required capabilities
+// for a specific resource type and HTTP method.
+type CapabilityChecker struct{}
+
+// NewCapabilityChecker creates a new instance of CapabilityChecker
+func NewCapabilityChecker() *CapabilityChecker {
+	return &CapabilityChecker{}
+}
+
+// ResourceType represents the type of resource being accessed
+type ResourceType string
+
+const (
+	ResourceTypeNode  ResourceType = "node"
+	ResourceTypeJob   ResourceType = "job"
+	ResourceTypeAgent ResourceType = "agent"
+	ResourceTypeOpen  ResourceType = "open"
+)
+
+// GetRequiredCapability determines the required capability for a specific resource type and HTTP method
+func (c *CapabilityChecker) GetRequiredCapability(resourceType ResourceType, method string) string {
+	isReadOperation := method == http.MethodGet
```

Comment:
should we add head as well? we don't have handler for head today, and we should just make sure we either return 404 or unaotharized access with `read:<resource>` capability required instead of saying `write:<resource>`